src/Reducers/index.js

```diff
@@ -0,0 +1,18 @@
+import { ADD_TODO, REMOVE_TODO, REMOVE_SELECTED, TOGGLE_SELECT } from '../Actions/types';
+
+const INITIAL_STATE = [];
+
+export default (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case ADD_TODO:
+      return action.payload.text === '' ? state : state.concat(action.payload);
+    case REMOVE_TODO:
+      return state.filter(todo => todo.id !== action.payload);
+    case REMOVE_SELECTED:
+      return state.filter(elem => elem.selected === false);
```

Comment:
`elem.selected` is already a boolean, you can safely do `elem => !elem.selected)`

Reply:
Great! I didn't know it works. Thanks for the tip!